modules/subscriptions/subscriptions/subscriptions-settings.tsx

```diff
@@ -90,6 +94,48 @@ const PaymentHistory: React.FC = () => {
   );
 };
 
+const CancelSubscriptonBtn = () => {
+  const { t } = useTranslation();
+  const [isOpen, setIsOpen] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const router = useRouter()
+  const onConfirm = async () => {
+    setIsLoading(true);
+    const data = await Service.require("subscriptions").cancelSubscription();
+    if(data.id) {
+      messageHandler.handleSuccess(t("Subscription cancelled successflly!"))
+      await router.push('/settings/subscriptions')
+    } else {
+      messageHandler.handleError(t("Failed to cancel subscription!"))
+    }
```

Comment:
Error shall be handled within the service it self. 

Because always we use the service we must handled the error (and if we do outside we may forget)

Reply:
Actually this is extra check, we have a check in the service too, but I am gonna remove this check and improve the service check.